frontend/src/concepts/pipelines/context/PipelinesContext.tsx

```diff
@@ -173,36 +185,149 @@ export const CreatePipelineServerButton: React.FC<CreatePipelineServerButtonProp
   const { namespace, refreshState } = React.useContext(PipelinesContext);
   const [creating, setCreating] = React.useState(false);
   const [error, setError] = React.useState<Error | null>(null);
+  const [configureModalVisible, setConfigureModalVisible] = React.useState(false);
+  const { project } = usePipelinesAPI();
 
-  const createCR = () => {
+  const createCR = (config: PipelineServerConfigType) => {
     setCreating(true);
     setError(null);
-    createPipelinesCR(namespace)
-      .then(() => {
-        refreshState();
-        // Don't reset creating state, this component is not needed once we have it and the caller
-        // should navigate away on their own as they will not need to create anymore
-      })
-      .catch((e) => {
-        setCreating(false);
-        setError(e);
-      });
+
+    Promise.all([
```

Comment:
I think all of this should be in another function outside of the Context file. We are aiming for js modules for private variables -- not full fledge functionality. Ideally delegate as much of this component out of this file so it can be easier maintained.

Reply:
I also imagine you want to dry run these first, no?

Reply:
Business logic really shouldn't be in the api folder -- it should really be for the thin layer that is API -- recommend a close-to-source util file for sure.